### ISSUE TITLE
Degraded nullable analysis with conditional expression of Generic type

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/roslyn#38556]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/content/problem/608446/degraded-nullable-analysis-with-conditional-expres.html)._

---
This code raised null reference warning at conditional expression assignment until on VS 16.2.0 Preview 1.0.
But now no warnings and happens NullReferenceException at runtime.
Please check below.

```csharp
#nullable enable
using System;

public class Program
{
    public static void Main()
    {
        var x = new Test<object>(new object());
        Console.WriteLine(x.Value.ToString()); // NullReferenceException here!
    }
}

public class Test<T>
{
    public T Value { get; }


    public Test(T value)
    {
        // this.Value = default; // this raises CS8602 correctly even at 16.2.0 Preview 2.0

        var dummyCond = value == null;
        this.Value = (dummyCond) ? value : default; // no nullable warnings since 16.2.0 Preview 2.0
    }
}
```

---
#### Visual Studio Feedback System on 6/17/2019, 03:47 AM: 

<p>We have directed your feedback to the appropriate engineering team for further evaluation. The team will review the feedback and notify you about the next steps.</p>

#### Visual Studio Feedback System on 6/17/2019, 08:14 PM: 

<p>This issue is currently being investigated. Our team will get back to you if either more information is needed, a workaround is available, or the issue is resolved.</p>

---
### Original Solutions
(no solutions)

**Comment by RikkiGibson** [comment]
This will warn if `default` is changed to `default(T)`. Therefore it seems like a dupe of #38339

**Comment by jcouv** [comment]
Tested this again in 16.5p3 and we produce a correct warning (passing `default(T)` value to a `T` property). I'll go ahead and close this. Thanks

![image](https://user-images.githubusercontent.com/12466233/73407136-093d2900-42ad-11ea-9aea-2b98db8ef3cd.png)
